.github/workflows/_ci.yaml

```diff
@@ -21,20 +21,107 @@ on:
         value: ${{ jobs.collect-docker-tags.outputs.TAGS }}
 
 permissions:
-  contents: read  # to fetch code
-  actions:  write # to cancel previous workflows
+  contents: read # to fetch code
+  actions: write # to cancel previous workflows
   packages: write # to upload container
 
 jobs:
+  pre-flight:
+    runs-on: ubuntu-22.04
+    outputs:
+      BUILD_TRITON: ${{ steps.run-conditions.outputs.BUILD_TRITON }}
+      BUILD_EQUINOX: ${{ steps.run-conditions.outputs.BUILD_EQUINOX }}
+      BUILD_MAXTEXT: ${{ steps.run-conditions.outputs.BUILD_MAXTEXT }}
+      BUILD_LEVANTER: ${{ steps.run-conditions.outputs.BUILD_LEVANTER }}
+      BUILD_T5X: ${{ steps.run-conditions.outputs.BUILD_T5X }}
+      BUILD_PAX: ${{ steps.run-conditions.outputs.BUILD_PAX }}
+      BUILD_GROK: ${{ steps.run-conditions.outputs.BUILD_GROK }}
+      TEST_JAX: ${{ steps.run-conditions.outputs.TEST_JAX }}
+      TEST_TRITON: ${{ steps.run-conditions.outputs.TEST_TRITON }}
+      TEST_MAXTEXT: ${{ steps.run-conditions.outputs.TEST_MAXTEXT }}
+      TEST_LEVANTER: ${{ steps.run-conditions.outputs.TEST_LEVANTER }}
+      TEST_UPSTREAM_PAX: ${{ steps.run-conditions.outputs.TEST_UPSTREAM_PAX }}
+      TEST_UPSTREAM_T5X: ${{ steps.run-conditions.outputs.TEST_UPSTREAM_T5X }}
+      TEST_ROSETTA_PAX: ${{ steps.run-conditions.outputs.TEST_ROSETTA_PAX }}
+      TEST_ROSETTA_T5X: ${{ steps.run-conditions.outputs.TEST_ROSETTA_T5X }}
+    steps:
+      - name: Check out the repository under ${GITHUB_WORKSPACE}
+        uses: actions/checkout@v4
+
+      - name: Validate input `TEST_SUBSET`
+        shell: bash
+        id: validate
+        run: |
+          test_subsets=$(echo "${{ inputs.TEST_SUBSET }}" | awk '{$1=$1;print}')
+          IFS=' ' read -a test_subsets <<< "$test_subsets"
+
+          valid_test_subsets=$(yq '.on.workflow_dispatch.inputs.TEST_SUBSET.options | join(",")'  .github/workflows/ci.yaml)
+          
+          for test_subset in "${test_subsets[@]}"; do     
+            if [[ $test_subset == "" || "$valid_test_subsets" != *"$test_subset"* ]]; then
+              message="Invalid value for '<TEST_SUBSET>' provided ❌. Expected one of: ($valid_test_subsets), Actual: '$test_subset'"
+              exit 1
+            fi
+          done
+
+          test_subsets=$(IFS=' '; echo "${test_subsets[*]}")
+          echo "test_subset=$test_subsets" >> "$GITHUB_OUTPUT"
+
+      - name: Compile run-conditions
+        id: run-conditions
+        shell: bash
+        run: |
+          if_testset_in() {
+            local list_of_items="$1"  # The list of items as a string
+            IFS=' ' read -a test_subsets <<< "${{ steps.validate.outputs.test_subset }}"
 
+            contains=0
+            for test_subset in "${test_subsets[@]}"; do     
+              if [[ $test_subset == "" || "$list_of_items" != *"$test_subset"* ]]; then
+                contains=$(( $contains | 0 ))
+              else
+                contains=$(( $contains | 1 ))
```

Comment:
You actually don't need to continue when you find it. Can you do something like this?
```
echo "true"
return
```
?

Reply:
good point!